java/src/main/native/src/RmmJni.cpp

```diff
@@ -280,7 +274,7 @@ class java_event_handler_memory_resource : public device_memory_resource {
     while (true) {
       try {
         total_before = tracker->get_total_allocated();
-        result       = resource->allocate(num_bytes, stream);
+        result       = resource->allocate(stream, num_bytes, rmm::CUDA_ALLOCATION_ALIGNMENT);
```

Comment:
No need to specify the default value here.
```suggestion
        result       = resource->allocate(stream, num_bytes);
```